pkg/controller/atlasdatabaseuser/atlasdatabaseuser_controller.go

```diff
@@ -127,7 +127,17 @@ func (r *AtlasDatabaseUserReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		return result.ReconcileResult(), nil
 	}
 
-	dus, err := dbuser.NewAtlasDatabaseUsersService(ctx, r.AtlasProvider, project.ConnectionSecretObjectKey(), log)
+	hasFallback := r.AtlasProvider.HasGlobalFallbackSecret()
+	credentialsSecret, err := customresource.ComputeSecretWithFallback(hasFallback, project, databaseUser)
```

Comment:
We should be making sure that we call out in the spec for independent CRs that the field for the local connection secret has ramifications on the non-independent case where it is being fallen back to if the global secret is not specified in the provider.

Reply:
will annotate that